### PR TITLE
Issue #7: Add HasItemPredicate which checks if there is an item in th…

### PR DESCRIPTION
**Pull request by mscibilia**
…e players inventory.

contribution to #7

### CODE REVIEW

**Comment by mscibilia** [comment]
@hhenrichsen Could you please add the "hacktoberfest-accepted" label to this PR also? Thanks!

**Comment by hhenrichsen** [comment]
I was thinking this should be looking for a *specific* ItemStack, rather than any itemstack. Can we make that change?

**Comment by hhenrichsen** [comment]
Additionally since it only occurs in one place, can we drop the `import static`?

**Comment by hhenrichsen** [comment]
@mscibilia I'm not sure if you have notifications on for this so I'm mentioning you to make sure you see this 🙂

File: src/main/java/com/github/shatteredsuite/core/commands/predicates/HasItemPredicate.java [hunk 1]
```diff
@@ -4,21 +4,22 @@
...
-import static java.util.Arrays.stream;
-
 public class HasItemPredicate extends SenderPlayerPredicate {
 
-    public HasItemPredicate(PredicateResponse response) {
+    private final int index;
+
+    public HasItemPredicate(PredicateResponse response, int index) {
         super(response);
+        this.index = index;
```
Comment (hhenrichsen) [inline src/main/java/com/github/shatteredsuite/core/commands/predicates/HasItemPredicate.java:13]: What does having this index accomplish? Are we looking in a certain inventory slot? I'd take an `ItemStack` as the parameter and search for that in the inventory.

File: src/main/java/com/github/shatteredsuite/core/commands/predicates/HasItemPredicate.java [hunk 2]
```diff
@@ -4,21 +4,22 @@
...
     }
 
     @Override
     public boolean test(CommandContext context) {
         boolean playerHasItem = false;
         if (super.test(context)) {
-            ItemStack[] playerInventoryContents = ((Player) context.sender).getInventory().getContents();
-            playerHasItem = stream(playerInventoryContents).anyMatch(itemStack -> itemStack.getAmount() > 0);
+            ItemStack itemStack = ((Player) context.sender).getInventory().getItem(index);
+            playerHasItem = itemStack != null && itemStack.getAmount() > 0;
```
Comment (hhenrichsen) [inline src/main/java/com/github/shatteredsuite/core/commands/predicates/HasItemPredicate.java:22]: As well down here, I'd use either `target.equals(itemStack)` or `target.isSimilar(itemStack)`.

**Review by hhenrichsen — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/shatteredsuite/core/commands/predicates/HasItemPredicate.java (hunk 1 above)
Comment (mscibilia) [reply src/main/java/com/github/shatteredsuite/core/commands/predicates/HasItemPredicate.java]: Sorry, I misunderstood the requirement. The change has been pushed

**Review by hhenrichsen — APPROVED** [review]

**Pull request merged**